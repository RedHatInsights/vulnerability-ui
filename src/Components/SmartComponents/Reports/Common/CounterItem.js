import React from 'react';
import { Text, View } from '@react-pdf/renderer';
import PropTypes from 'prop-types';

import styles from './styles';

const CounterItem = ({ counter, label }) => {
    return (
        <View>
            <Text style={styles.counter}>
                {counter}
            </Text >
            <Text style={styles.counterLabel}>
                {label}
            </Text>
        </View>
    );
};

CounterItem.propTypes = {
    counter: PropTypes.number,
    label: PropTypes.string
};

export default CounterItem;
