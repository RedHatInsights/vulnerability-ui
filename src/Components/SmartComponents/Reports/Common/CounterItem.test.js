import { render } from '@testing-library/react';
import CounterItem from './CounterItem';

describe('CounterItem component', () => {
    it('Should match snapshot.', () => {
        const { asFragment } = render(
            <CounterItem counter={100} label="items" />
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
