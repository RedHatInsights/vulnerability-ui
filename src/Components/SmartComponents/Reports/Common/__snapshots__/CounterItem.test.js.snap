// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterItem component Should match snapshot. 1`] = `
<CounterItem
  counter={100}
  label="items"
>
  <VIEW>
    <TEXT
      style={
        Object {
          "color": "#C9190B",
          "fontSize": 20,
        }
      }
    >
      100
    </TEXT>
    <TEXT
      style={
        Object {
          "fontSize": 9,
        }
      }
    >
      items
    </TEXT>
  </VIEW>
</CounterItem>
`;
